sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/stream/PrefetchableIterable.java

```diff
@@ -20,6 +20,9 @@
 /** An {@link Iterable} that returns {@link PrefetchableIterator}s. */
 public interface PrefetchableIterable<T> extends Iterable<T> {
 
+  /** Ensures that the next iterator returned has been prefetched. */
+  void prefetch();
```

Comment:
It was errorprone with the previous setup where we were effectively discarding the prefetched iterator making it such that the iterable needed to handle the caching. This seems to be much safer since the iterable is usually stored.